core/lib/multistore.ts

```diff
@@ -215,5 +215,19 @@ export function localizedRouteConfig (route: RouteConfig, storeCode: string, isC
 export function localizedRoutePath (path: string, storeCode: string): string {
   const _path = path.startsWith('/') ? path.slice(1) : path
 
-  return `/${storeCode}/${_path}`
+  return `${storeCodeToStoreUrl(storeCode)}/${_path}`
+}
+
+export function storeCodeToStoreUrl (storeCode: string): string {
+  const store = config.storeViews && config.storeViews[storeCode]
+
+  if (!store) {
+    return ''
+  }
+
+  if (store.url) {
+    return store.url
+  }
+
+  return store.appendStoreCode ? `/${storeCode}` : ''
```

Comment:
This flag probably needs to be checked even for appending `url` - moreover we're not checking here if the urrl contains `https://` neither `http://` which can possible happen 

@lukeromanowicz you should take a look at it